homeassistant/components/yamaha_musiccast/media_player.py

```diff
@@ -611,38 +626,39 @@ async def async_join_players(self, group_members):
             if self.is_server
             else uuid.random_uuid_hex().upper()
         )
+
+        ip_addresses = set()
         # First let the clients join
         for client in entities:
             if client != self:
                 try:
-                    await client.async_client_join(group, self)
+                    if await client.async_client_join(group, self):
+                        ip_addresses.add(client.ip_address)
```

Comment:
Please only wrap the line that can raise in the try... except block. Use and `else:` block if needed.